prereise/gather/griddata/hifld/data_process/generators.py

```diff
@@ -132,5 +191,11 @@ def build_plant(bus, substations):
         ["Summer Capacity (MW)", "Winter Capacity (MW)"]
     ].max(axis=1)
     generators.rename({"Minimum Load (MW)": "Pmin"}, inplace=True, axis=1)
+    print("Fitting heat rate curves to EPA data... (this may take several minutes)")
+    heat_rate_curve_estimates = generators.apply(
+        lambda x: estimate_heat_rate_curve(x, epa_ampd_groupby, crosswalk_translation),
+        axis=1,
+    )
+    generators = generators.join(heat_rate_curve_estimates)
```

Comment:
To keep the user facing function slim we could move the above (what is under the **Add information** comment) into a `add_information_to_generators` function. Just a suggestion.

Reply:
Not a bad idea. I'd lean toward waiting until we have most of the logic built in, then we can better decide how to group and delegate.